sycl/doc/extensions/experimental/sycl_ext_oneapi_memory_export.asciidoc

```diff
@@ -0,0 +1,378 @@
+represented by a `void *`.
+
+The `exported_mem` template struct is specialized for each of the
+`external_mem_handle_type` values. Each specialization defines a `type` alias
+that represents the type of the exported memory handle for that specific handle.
+
+The `exported_mem_t` type alias allows the user to easily obtain the type of
+the exported memory handle based on the `external_mem_handle_type` template
+parameter.
+
+=== API of the extension
+
+```c++
+
+namespace sycl::ext::oneapi::experimental {
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    external_mem_handle_type externalMemHandleType,
+    const sycl::device &syclDevice, const sycl::context &syclContext,
+    const property_list& propList = {});
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    external_mem_handle_type externalMemHandleType,
+    const sycl::queue &syclQueue,
+    const property_list& propList = {});
+}
+```
+
+The `alloc_exportable_device_mem` function allocates memory on the device marked
+as having the ability to later export that memory to an external memory resource
+handle.
+
+Memory allocated through this function must only be freed using
+`free_exportable_mem`. Using `sycl::free` to deallocate memory allocated with
+this function results in undefined behavior.
+
+With the exception of the `sycl::free` function from the core SYCL
+specification, pointers to memory allocated through this function may be passed
+to any core SYCL specification API accepting device USM memory pointers.
+
+Memory allocated through this function is only available on device.
+
+Memory allocated through this function has a linear memory layout on the device 
+(which is the same as memory allocated by other USM allocation functions like 
+`sycl::malloc_device`).
+
+Zero or more properties can be passed in the `propList` parameter via an
+instance of `sycl::property_list`. Currently, this extension does not define
+any properties that can be used with this function, so the `propList` parameter
+is ignored and reserved for future use.
+
+The passed `externalMemHandleType` must be supported by the operating system
+running the SYCL application. Passing an `externalMemHandleType` that is not
+supported by the operating system running the SYCL application results in
+undefined behavior. The `win32_nt_handle` handle type is only supported on
+Windows operating systems, while the `opaque_fd` handle type is only supported
+on POSIX compliant operating systems.
+
+This function will throw a `sycl::exception` with `errc::feature_not_supported`
+if the device `syclDevice` does not have
+`aspect::ext_oneapi_exportable_device_mem`.
+
+This function will throw a `sycl::exception` with the `errc::runtime` code if
+any error occurs while allocating the memory.
+
+This function will throw a `sycl::exception` with the code `errc::invalid` if
+the `externalMemHandleType` passed is not supported by the backend
+implementation.
```

Comment:
This is too vague.  Users of the API should have some guarantee about which values of `externalMemHandleType` are supported.  I suggest combining this paragraph with the one above that starts "The passed `externalMemHandleType` must be ..." into something like:

> Only certain values of `externalMemHandleType` are supported:
>
> * `external_mem_handle_type::opaque_fd` is supported when the host is a Posix compliant operating system.
> * `external_mem_handle_type::win32_nt_handle` is supported when the host is Windows.
>
> No other values are supported.  Passing an unsupported value causes this function to throw an `exception` with the `errc::feature_not_supporterd` error code.

Reply:
Applied your suggestion.